keras_cv/applications/object_detection/retina_net/retina_net.py

```diff
@@ -0,0 +1,323 @@
+
+        with tf.GradientTape() as tape:
+            predictions = self(x, training=True)
+            # predictions technically do not have a format
+            # loss accepts
+            loss = self.compiled_loss(
+                y_training_target,
+                predictions["train_preds"],
+                regularization_losses=self.losses,
+            )
+
+        # Training specific code
+        trainable_vars = self.trainable_variables
+        gradients = tape.gradient(loss, trainable_vars)
+        self.optimizer.apply_gradients(zip(gradients, trainable_vars))
+
+        # To minimize GPU transfers, we update metrics AFTER we take grades and apply
+        # them.
+
+        # TODO(lukewood): assert that all metric formats are the same
+        self._update_metrics(y_for_metrics, predictions["inference"])
+        return self._metrics_result(loss)
+
+    def test_step(self, data):
+        x, y = data
+        y_for_metrics, y_training_target = self._encode_data(x, y)
+
+        predictions = self(x)
+        loss = self.compiled_loss(
+            y_training_target,
+            predictions["train_preds"],
+            regularization_losses=self.losses,
+        )
+
+        self._update_metrics(y_for_metrics, predictions["inference"])
+        return self._metrics_result(loss)
+
+    def _update_metrics(self, y_true, y_pred):
+        y_true = bounding_box.convert_format(
+            y_true,
+            source=self.bounding_box_format,
+            target=self._metrics_bounding_box_format,
+        )
+        y_pred = bounding_box.convert_format(
+            y_pred,
+            source=self.bounding_box_format,
+            target=self._metrics_bounding_box_format,
+        )
+        self.compiled_metrics.update_state(y_true, y_pred)
+
+    def _metrics_result(self, loss):
+        metrics_result = {m.name: m.result() for m in self.metrics}
+        metrics_result["loss"] = loss
+        return metrics_result
+
+    def inference(self, x):
+        predictions = self.predict(x)
+        return predictions["inference"]
+
+
+def _parse_backbone(backbone, include_rescaling, backbone_weights):
+    if isinstance(backbone, str):
+        if backbone == "resnet50":
+            return _resnet50_backbone(include_rescaling, backbone_weights)
+        else:
+            raise ValueError(
+                "backbone expected to be one of ['resnet50', keras.Model]. "
+                f"Received backbone={backbone}."
+            )
+    if include_rescaling or backbone_weights:
+        raise ValueError(
+            "When a custom backbone is used, include_rescaling and "
+            f"backbone_weights are not supported.  Received backbone={backbone}, "
+            f"include_rescaling={include_rescaling}, and "
+            f"backbone_weights={backbone_weights}."
+        )
+    if not callable(backbone):
+        raise ValueError(
+            "Custom backbones should be subclasses of a keras.Model. "
```

Comment:
If we are expecting a keras model instance, should we just use isintance() here above?